Das Upgrade von curl auf die Version 7.53 erfordert die Anpassung eines OpenWrt-Patchs.
--- a/openwrt/package/network/utils/curl/patches/102-CVE-2017-7468.patch
+++ b/openwrt/package/network/utils/curl/patches/102-CVE-2017-7468.patch
@@ -84,8 +84,8 @@ Reported-by: lijian996@users.noreply.git
    /* Put our freshly minted SSL session in cache */
 -  if(data->set.general_ssl.sessionid) {
 +  if(SSL_SET_OPTION(primary.sessionid)) {
-     const uint8_t *ssl_sessionid = ssl_get_session_id_size(ssl);
-     size_t ssl_idsize = ssl_get_session_id(ssl);
+     const uint8_t *ssl_sessionid = ssl_get_session_id(ssl);
+     size_t ssl_idsize = ssl_get_session_id_size(ssl);
      Curl_ssl_sessionid_lock(conn);
 --- a/lib/vtls/cyassl.c
 +++ b/lib/vtls/cyassl.c
--- a/openwrt/package/network/utils/curl/patches/106-CVE-2017-8817.patch
+++ b/openwrt/package/network/utils/curl/patches/106-CVE-2017-8817.patch
@@ -73,7 +73,7 @@ index dc1cc03bc..6eb37d81d 100644
 @@ -121,6 +121,7 @@
  test1128 test1129 test1130 test1131 test1132 test1133 test1134 test1135 \
  test1136 test1137 test1138 test1139 test1140 test1141 test1142 test1143 \
- test1144 \
+ test1144 test1145 test1146 \
 +test1163 \
  test1200 test1201 test1202 test1203 test1204 test1205 test1206 test1207 \
  test1208 test1209 test1210 test1211 test1212 test1213 test1214 test1215 \
--- a/openwrt/package/network/utils/curl/patches/320-mbedtls-nonblocking-handshake.patch
+++ /dev/null
@@ -1,27 +0,0 @@
-From b993d2cca536870ecdf3b4611de9f77215af8eb8 Mon Sep 17 00:00:00 2001
-From: Antoine Aubert <a.aubert@overkiz.com>
-Date: Fri, 20 Jan 2017 08:10:28 +0100
-Subject: [PATCH] vtls: fix mbedtls multi non blocking handshake.
-
-When using multi, mbedtls handshake is in non blocking mode.
-vtls must set wait for read/write flags for the socket.
----
- lib/vtls/vtls.c | 5 +++--
- 1 file changed, 3 insertions(+), 2 deletions(-)
-
-diff --git a/lib/vtls/vtls.c b/lib/vtls/vtls.c
-index fad9335bbf..871622fef1 100644
---- a/lib/vtls/vtls.c
-+++ b/lib/vtls/vtls.c
-@@ -485,8 +485,9 @@ void Curl_ssl_close_all(struct Curl_easy *data)
- }
- 
- #if defined(USE_OPENSSL) || defined(USE_GNUTLS) || defined(USE_SCHANNEL) || \
--    defined(USE_DARWINSSL) || defined(USE_NSS)
--/* This function is for OpenSSL, GnuTLS, darwinssl, and schannel only. */
-+    defined(USE_DARWINSSL) || defined(USE_NSS) || defined(USE_MBEDTLS)
-+/* This function is for OpenSSL, GnuTLS, darwinssl, mbedtls, and schannel
-+   only. */
- int Curl_ssl_getsock(struct connectdata *conn, curl_socket_t *socks,
-                      int numsocks)
- {
